shared/types/node.d.ts

```diff
@@ -14,13 +19,18 @@ interface CustomNodeConfig {
 }
 
 interface StaticNodeConfig {
+  isCustom: false;
```

Comment:
This is used for type guards throughout the codebase so TS knows what node config its dealing with.